ci-operator/step-registry/hypershift/mce/power/create/ibmcloud-workers/lb/hypershift-mce-power-create-ibmcloud-workers-lb-commands.sh

```diff
@@ -0,0 +1,178 @@
+export HOSTED_CONTROL_PLANE_NAMESPACE="${CLUSTERS_NAMESPACE}-${HOSTED_CLUSTER_NAME}"
+
+# Fetching Domain from Vault
+if [[ -z "${HYPERSHIFT_BASE_DOMAIN}" ]]; then
+		HYPERSHIFT_BASE_DOMAIN=$(jq -r '.hypershiftBaseDomain' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources.json")
+fi
+
+# Fetching Resource Group Name from Vault
+if [[ -z "${RESOURCE_GROUP_NAME}" ]]; then
+    RESOURCE_GROUP_NAME=$(jq -r '.resourceGroupName' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources-het.json")
+fi
+
+# Fetching CIS related details from Vault
+if [[ -z "${CIS_INSTANCE}" ]]; then
+		CIS_INSTANCE=$(jq -r '.cisInstance' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources.json")
+fi
+if [[ -z "${CIS_DOMAIN_ID}" ]]; then
+		CIS_DOMAIN_ID=$(jq -r '.cisDomainID' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources.json")
+fi
+
+# Installing generic required tools
+echo "$(date) Installing required tools"
+mkdir /tmp/ibm_cloud_cli
+curl --output /tmp/IBM_CLOUD_CLI_amd64.tar.gz https://download.clis.cloud.ibm.com/ibm-cloud-cli/2.16.1/IBM_Cloud_CLI_2.16.1_amd64.tar.gz
+tar xvzf /tmp/IBM_CLOUD_CLI_amd64.tar.gz -C /tmp/ibm_cloud_cli
+export PATH=${PATH}:/tmp/ibm_cloud_cli/Bluemix_CLI/bin
+mkdir /tmp/bin
+curl -L https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 -o /tmp/bin/jq && chmod +x /tmp/bin/jq
+export PATH=$PATH:/tmp/bin
+
+echo | ibmcloud login --apikey @"${AGENT_POWER_CREDENTIALS}/ibmcloud-apikey"
+ibmcloud target -g "${RESOURCE_GROUP_NAME}"
+
+# Installing required ibmcloud plugins
+echo "$(date) Installing required ibmcloud plugins"
+ibmcloud plugin install vpc-infrastructure
+ibmcloud plugin install cis
+ibmcloud cis instance-set ${CIS_INSTANCE}
+
+set_lb_configs() {
+	# LoadBalancer configs
+	LB_NAME="lb-${HOSTED_CLUSTER_NAME}"
+	LB_ID=""
+
+	# Other resource configs
+	IP_X86=$(cat ${SHARED_DIR}/ipx86addr)
+	IP_POWER=$(cat ${SHARED_DIR}/ippoweraddr)
+
+	# Fetch VPC details
+	if [[ -z "${VPC_REGION}" ]]; then
+		VPC_REGION=$(jq -r '.vpcRegion' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources-het.json")
+	fi
+	if [[ -z "${VPC_NAME}" ]]; then
+		VPC_NAME=$(jq -r '.vpcName' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources-het.json")
+	fi
+	if [[ -z "${VPC_SUBNET_ID}" ]]; then
+		VPC_SUBNET_ID=$(jq -r '.vpcSubnetID' "${AGENT_POWER_CREDENTIALS}/ibmcloud-resources-het.json")
+	fi
+}
+
+dns_entry() {
+	# Create DNS entry
+	LB_ID=$1
+
+	# Clean up the record previously created with worker IP.
+	idToDelete=$(ibmcloud cis dns-records ${CIS_DOMAIN_ID} --name "*.apps.${HOSTED_CLUSTER_NAME}.${HYPERSHIFT_BASE_DOMAIN}" --output json | jq -r '.[].id')
+	if [ -n "${idToDelete}" ]; then
+		ibmcloud cis dns-record-delete ${CIS_DOMAIN_ID} ${idToDelete}
+	fi
+
+	LB_HOSTNAME=$(ibmcloud is load-balancer "${LB_ID}" --output JSON | jq -r '.hostname')
+	# Create a new record with lb
+	ibmcloud cis dns-record-create ${CIS_DOMAIN_ID} --type CNAME --name "*.apps.${HOSTED_CLUSTER_NAME}" --content "${LB_HOSTNAME}"
+}
+
+wait_lb_active() {
+	INTERVAL=$1
+	while true; do
```

Comment:
Maybe you want to add some kind of timeout?

Reply:
Added it, thanks!